exporter/tanzuobservabilityexporter/exporter.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tanzuobservabilityexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/google/uuid"
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultApplicationName = "defaultApp"
+	defaultServiceName     = "defaultService"
+	labelApplication       = "application"
+	labelError             = "error"
+	labelEventName         = "name"
+	labelService           = "service"
+	labelSpanKind          = "span.kind"
+	labelStatusMessage     = "status.message"
+	labelStatusCode        = "status.code"
+)
+
+// spanSender Interface for sending tracing spans to Tanzu Observability
+type spanSender interface {
+	// SendSpan mirrors sender.SpanSender from wavefront-sdk-go.
+	// traceId, spanId, parentIds and preceding spanIds are expected to be UUID strings.
+	// parents and preceding spans can be empty for a root span.
+	// span tag keys can be repeated (example: "user"="foo" and "user"="bar")
+	SendSpan(name string, startMillis, durationMillis int64, source, traceID, spanID string, parents, followsFrom []string, tags []senders.SpanTag, spanLogs []senders.SpanLog) error
+	Flush() error
+	Close()
+}
+
+type tracesExporter struct {
+	cfg    *Config
+	sender spanSender
+	logger *zap.Logger
+}
+
+func newTracesExporter(l *zap.Logger, c config.Exporter) (*tracesExporter, error) {
+	cfg, ok := c.(*Config)
+	if !ok {
+		return nil, fmt.Errorf("invalid config: %#v", c)
+	}
+
+	endpoint, err := url.Parse(cfg.Traces.Endpoint)
+	if nil != err {
+		return nil, fmt.Errorf("failed to parse traces endpoint: %v", err)
+	}
+	tracingPort, err := strconv.Atoi(endpoint.Port())
+	if nil != err {
+		return nil, fmt.Errorf("failed to parse traces port: %v", err)
+	}
+
+	s, err := senders.NewProxySender(&senders.ProxyConfiguration{
+		Host:                 endpoint.Hostname(),
+		MetricsPort:          2878,
```

Comment:
I guess this is not used since you are not sending metrics. In that case, can you add a comment about it?

Reply:
The Wavefront SDK sends internal metrics about its own operations via this port - I'll add a comment.

> It would also help if someone from your company reviewed this PR too (not a strong requirement AFAIK).

@oppegard and I paired on this, and it's all been reviewed and accepted from our side.

I'll take a look at where we're missing coverage and add tests as appropriate (we tend to follow TDD and haven't really looked at code coverage, so thanks for the tip!)